lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflectData.java

```diff
@@ -76,6 +78,25 @@ void genericProtocol() {
     assertThat(existsArgument.schema(), equalTo(Schema.create(Schema.Type.STRING)));
   }
 
+  @Test
+  void fieldsOrder() {
+    Schema schema = ReflectData.get().getSchema(Meta.class);
+    List<Schema.Field> fields = schema.getFields();
+    assertEquals(fields.size(), 4);
+    assertEquals(fields.get(0).name(), "f1");
+    assertEquals(fields.get(1).name(), "f2");
+    assertEquals(fields.get(2).name(), "f3");
+    assertEquals(fields.get(3).name(), "f4");
+
+    schema = new ReflectData(false).getSchema(Meta.class);
+    fields = schema.getFields();
+    assertEquals(fields.size(), 4);
+    assertEquals(fields.get(0).name(), "f1");
+    assertEquals(fields.get(1).name(), "f4");
+    assertEquals(fields.get(2).name(), "f2");
+    assertEquals(fields.get(3).name(), "f3");
```

Comment:
This order depends on the JVM in use. Can we please ensure this test is only enabled for JVMs where the test is known to succeed?

Reply:
addressed